staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -53,10 +53,31 @@ var (
 		},
 		[]string{"resource"},
 	)
+
+	watcherInputBufferSize = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name: "apiserver_watcher_input_buffer_size",
+			Help: "Watcher buffer size distribution for each resource.",
+			// There are two types of buffers {10, 1000}. Thus we customize buckets
```

Comment:
If all the values are either 10 or 1000, then the histogram would record counts in only the first,second and last (+Inf) bucket, so I don't think there is much utility in using a histogram here. I think it would make more sense to use a counter with buffer_size as a label since the values are well-defined and constrained. @brancz to confirm though, I could be mistaken.

Reply:
Agree on adding buffer_size as a label to distinguish the two types. But a counter can not tell the pressure clearly of all watchers in memory.

Reply:
A histogram is already just a collection of counters. 

Let's say you have a resource 'a' and the watcher input buffer size is 10. Then your metrics will look something like this:

`
apiserver_watcher_input_buffer_size{resource="a",le="9"} 0
apiserver_watcher_input_buffer_size{resource="a",le=“10”} 1
apiserver_watcher_input_buffer_size{resource="a",le="50"} 1
apiserver_watcher_input_buffer_size{resource="a",le=“999”} 1
apiserver_watcher_input_buffer_size{resource="a",le="+Inf"} 1
`

Let's say you have another observation of resource 'a' with a buffer size of 1000. Then, the metrics will now look something like this:

`
apiserver_watcher_input_buffer_size{resource="a",le="9"} 0
apiserver_watcher_input_buffer_size{resource="a",le=“10”} 1
apiserver_watcher_input_buffer_size{resource="a",le="50"} 1
apiserver_watcher_input_buffer_size{resource="a",le=“999”} 1
apiserver_watcher_input_buffer_size{resource="a",le="+Inf"} 2
`

The first bucket will always be zero since you will presumably only ever have buffer input size of 10 or 1000 and it will count observations which are less than or equal to 9. Each observation of a resource with buffer input size of 10 will increment all subsequent buckets. Each observation of a resource with buffer input size of 1000 will increment only the last bucket, since each previous bucket compares to a size of 999 or smaller.

Reply:
If it is true that all the buffer sizes are either 10 or 1000, then I don't believe that a histogram is the best choice here. But again, that is based on my understanding of metric values based on your comment.

Reply:
sorry, 10 or 1000 is capacity of the buffer, what we want to know is the size distribution.

Reply:
Okay. Then (at least to me) that comment was easy to misunderstand, maybe it would be worth modifying it to be clearer?

Also, the bucket sizes don't really seem to make sense to me. There is only two levels of granularity for buffers which are smaller than 10 (i.e. 9 & 10) and it seems weird to only ambiguate those two specifically. Wouldn't it make sense to have slightly better granularity to measure the distributions of buffer sizes for the smaller capacity buffers? Why 9 and 10 buckets, specifically?